### ISSUE TITLE
UnitAlgebra BestSI for time

### ISSUE THREAD

**Issue by bjmoor** [sstsimulator/sst-core#305]
It would be nice to special-case the BestSI functions on UnitAlgebra for time, specifically when the amount of time passed is greater than seconds.   Most people don't typically think in KiloSeconds and MegaSeconds, but rather minutes, hours, days, etc.

**Comment by jjwilke** [comment]
I agree that Ks and Ms are weird units.  I worry that this might break any string parsing code out there - and might make plotting results MORE difficult. If I grab a time from the file, I don't want to have to parse hours + minutes + seconds and convert to seconds.

What if we print us, ms, s - then stop at seconds and just print larger seconds. In parentheses, we could print hours, minutes, seconds for human readable but still keep seconds available for parsing and avoid breaking anyone's parsing scripts.

````
1.3456Ks -> 1345.6s (22m 25.6s)
````

**Comment by feldergast** [comment]
This is a challenge to do since UnitAlgebra doesn't really know anything about the units that are registered with it.  We could possibly change the unit registration to include the ability to have functions that can do some of these conversions.  I'll look into it.
